### PR TITLE
+tck #236 example subscriber whitebox tested, and whitebox fixed

### PR DESCRIPTION
**Pull request by ktoso**
Adds better example on using `SubscriberWhiteboxVerification` - it now explains how you would use it with your impl, thanks @purplefox for pointing out the need for clarification here in #236!

This PR also makes us use the whitebox to test the example subscriber. This found a bug in the new tests introduced during https://github.com/reactive-streams/reactive-streams-jvm/pull/209/files as it accidentally subscribed twice (whitebox's Stage is subscribing the `sub` to the `pub` during creation). Fixed tests, and improved docs on these methods.

Resolves https://github.com/reactive-streams/reactive-streams-jvm/issues/236

### CODE REVIEW

**Comment by ktoso** [comment]
// cc @viktorklang

File: tck/src/main/java/org/reactivestreams/tck/SubscriberWhiteboxVerification.java [hunk 1]
```diff
@@ -560,6 +553,10 @@ public SubscriberPuppet puppet() {
 
     public T signalNext() throws InterruptedException {
       T element = nextT();
+      return signalNext(element);
+    }
+
+    public T signalNext(T element) throws InterruptedException {
```
Comment (viktorklang) [inline tck/src/main/java/org/reactivestreams/tck/SubscriberWhiteboxVerification.java]: Public?

File: tck/src/main/java/org/reactivestreams/tck/SubscriberWhiteboxVerification.java [hunk 2]
```diff
@@ -560,6 +553,10 @@ public SubscriberPuppet puppet() {
 
     public T signalNext() throws InterruptedException {
       T element = nextT();
+      return signalNext(element);
```
Comment (viktorklang) [inline tck/src/main/java/org/reactivestreams/tck/SubscriberWhiteboxVerification.java]: `return signalNext(nextT())`?

**Comment by viktorklang** [comment]
Aside from comments, :+1:!

File: tck/src/main/java/org/reactivestreams/tck/SubscriberWhiteboxVerification.java (hunk 1 above)
Comment (ktoso) [inline tck/src/main/java/org/reactivestreams/tck/SubscriberWhiteboxVerification.java]: can be private, yeap

File: tck/src/main/java/org/reactivestreams/tck/SubscriberWhiteboxVerification.java (hunk 2 above)
Comment (ktoso) [inline tck/src/main/java/org/reactivestreams/tck/SubscriberWhiteboxVerification.java]: sure

**Comment by ktoso** [comment]
Comments addresses :+1:

**Comment by viktorklang** [comment]
@reactive-streams/contributors this is a documentation fix and bugfix for the TCK, will merge in 48h if no objections have been made

**Comment by benjchristensen** [comment]
Since this is all in the TCK I'm assuming it's fixing conformance with the spec, so I'm okay with it.

**Pull request merged**